src/placeos-driver/proxy/remote_driver.cr

```diff
@@ -178,31 +180,10 @@ module PlaceOS::Driver::Proxy
       module_id = module_id?
       raise Error.new(ErrorCode::ModuleNotFound, "could not find module id", *@error_details) unless module_id
 
-      core_uri = which_core(module_id)
-
-      # build request
-      core_uri.path = "/api/core/v1/command/#{module_id}/execute"
-      response = HTTP::Client.post(
-        core_uri,
-        headers: HTTP::Headers{"X-Request-ID" => request_id || UUID.random.to_s},
-        body: {
-          "__exec__" => function,
-          function   => args || named_args,
-        }.to_json
-      )
+      exec_args = args || named_args
 
-      case response.status_code
-      when 200
-        # exec was successful, json string returned
-        response.body
-      when 203
-        # exec sent to module and it raised an error
-        info = NamedTuple(message: String, backtrace: Array(String)?).from_json(response.body)
-        raise Error.new(ErrorCode::RequestFailed, "module raised: #{info[:message]}", *@error_details, info[:backtrace])
-      else
-        # some other failure 3
-        raise Error.new(ErrorCode::UnexpectedFailure, "unexpected response code #{response.status_code}", *@error_details)
-      end
+      client = Core::Client.new(which_core)
+      client.execute(module_id, function, exec_args)
```

Comment:
Is the client doing exactly the same stuff that was previously happening here? Doesn't look like the request ID is being forwarded